File: Modules/Groups/Models/Group.php
<?php

namespace Modules\Groups\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;
use Modules\Groups\Enums\GroupStatus;
use Modules\Groups\Enums\GroupType;
use Modules\Sessions\Models\Session;
use Modules\User\Models\User;

class Group extends Model
{
    use SoftDeletes;

    protected $fillable = [
        'instructor_id',
        'substance_category_id',
        'group_type',
        'status',
        'name_ar',
        'name_en',
        'min_members',
        'max_members',
    ];

    protected $casts = [
        'group_type' => GroupType::class,
        'status' => GroupStatus::class,
    ];

    public function instructor(): BelongsTo
    {
        return $this->belongsTo(User::class, 'instructor_id');
    }

    public function members(): BelongsToMany
    {
        return $this->belongsToMany(User::class, 'group_members')
            ->withPivot(['joined_at', 'is_active'])
            ->wherePivot('is_active', 1);
    }

    public function sessions(): HasMany
    {
        return $this->hasMany(Session::class);
    }

    public function getMemberCountAttribute(): int
    {
        return $this->members()->count();
    }
}

File: Solid.Api/Database/Entities/Group.cs
namespace Solid.Api.Database.Entities;

public sealed class Group
{
    public long Id { get; set; }

    public long? InstructorId { get; set; }

    public long? SubstanceCategoryId { get; set; }

    public string GroupType { get; set; } = string.Empty;

    public string Status { get; set; } = string.Empty;

    public string NameAr { get; set; } = string.Empty;

    public string NameEn { get; set; } = string.Empty;

    public byte MinMembers { get; set; } = 7;

    public byte MaxMembers { get; set; } = 15;

    public DateTime? DeletedAt { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public User? Instructor { get; set; }

    public SubstanceCategory? SubstanceCategory { get; set; }

    public ICollection<GroupMember> Members { get; set; } = [];

    public ICollection<TherapySession> Sessions { get; set; } = [];
}
